docs/book/src/tasks/experimental-features/cluster-classes.md

```diff
@@ -0,0 +1,497 @@
+              effect: NoSchedule
+          serviceAccountName: kindnet
+          containers:
+            - name: kindnet-cni
+              image: kindest/kindnetd:0.5.4
+              env:
+                - name: HOST_IP
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: status.hostIP
+                - name: POD_IP
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: status.podIP
+                - name: POD_SUBNET
+                  value: '${DOCKER_POD_CIDRS}'
+              volumeMounts:
+                - name: cni-cfg
+                  mountPath: /etc/cni/net.d
+                - name: xtables-lock
+                  mountPath: /run/xtables.lock
+                  readOnly: false
+                - name: lib-modules
+                  mountPath: /lib/modules
+                  readOnly: true
+              resources:
+                requests:
+                  cpu: "100m"
+                  memory: "50Mi"
+                limits:
+                  cpu: "100m"
+                  memory: "50Mi"
+              securityContext:
+                privileged: false
+                capabilities:
+                  add: ["NET_RAW", "NET_ADMIN"]
+          volumes:
+            - name: cni-cfg
+              hostPath:
+                path: /etc/cni/net.d
+            - name: xtables-lock
+              hostPath:
+                path: /run/xtables.lock
+                type: FileOrCreate
+            - name: lib-modules
+              hostPath:
+                path: /lib/modules
+---
+apiVersion: addons.cluster.x-k8s.io/v1alpha4
+kind: ClusterResourceSet
+metadata:
+  name: cni
+spec:
+  strategy: ApplyOnce
+  clusterSelector:
+    matchLabels:
+      cni: kindnet
+  resources:
+    - name: kindnet
+      kind: ConfigMap
+```
+
+Copy the above yaml into a local text file - kindnet.yaml - and run:
+```bash
+kubectl apply -f kindnet.yaml
+```
+
+#### Apply the workload cluster
+The below is a Cluster definition that leverages the ClusterClass created above to define its shape.
+```yaml
+apiVersion: cluster.x-k8s.io/v1alpha4
+kind: Cluster
+metadata:
+  name: "clusterclass-quickstart"
+  namespace: default
+  labels:
+    cni: kindnet
+spec:
+  clusterNetwork:
+    services:
+      cidrBlocks: ["10.128.0.0/12"]
+    pods:
+      cidrBlocks: ["192.168.0.0/16"]
+    serviceDomain: "cluster.local"
+  topology:
+    class: cluster-class
+    version: v1.21.2
+    controlPlane:
+      metadata:
+        labels:
+          topologyLabel: topologyLabelValue
+        annotations:
+          topologyAnnotation: topologyAnnotation
+      replicas: 1 
+    workers:
+      machineDeployments:
+        - class: linux-worker
+          name: linux-workers
+          metadata:
+            labels:
+              topologyLabel: topologyLabelValue
+            annotations:
+              topologyAnnotation: topologyAnnotation
+          replicas: 1 
```

Comment:
```suggestion
spec:
  topology:
    class: cluster-class
    version: v1.21.2
    controlPlane:
      replicas: 1 
    workers:
      machineDeployments:
        - class: linux-worker
          name: linux-workers
          replicas: 1 
```

Should we shorten the YAML to this? Imho it shows better how simple it is to create a Cluster with ClusterClass

Reply:
Looks like the simplest thing to do is to keep the kindnet config but get rid of the labels.

Reply:
Kindnet needs the values in the clusterNetwork it seems or else the nodes can't come up - is there somewhere else I can set them to get around this?

Reply:
I have no idea how kindnet is configured :). But I think it's fine to leave it when it's required by kindnet.